### PR TITLE
Add ADC driver for EFM32 

### PR DESCRIPTION
**Pull request by wsipak**
This adds ADC driver for Silicon Labs Gecko Series 1 ADC, and ADC node using this driver for the `efm32pg_stk3402a` board.

### CODE REVIEW

**Review by fkokosinski — DISMISSED** [review]
LGTM

File: soc/arm/silabs_exx32/efm32pg1b/Kconfig.series [hunk 1]
```diff
@@ -18,5 +18,6 @@ config SOC_SERIES_EFM32PG1B
 	select SOC_GECKO_CMU
```
Comment (DRuffer-tmo) [inline soc/arm/silabs_exx32/efm32pg1b/Kconfig.series:18]: Shouldn't this be in soc/arm/silabs_exx32/efm32pg12b/Kconfig.series?
The boards/arm/efm32pg_stk3402a/efm32pg_stk3402a.dts has #include <silabs/efm32pg12b500f1024gl125.dtsi>.

**Review by DRuffer-tmo — CHANGES_REQUESTED** [review]

File: soc/arm/silabs_exx32/efm32pg1b/Kconfig.series (hunk 1 above)
Comment (wsipak) [reply soc/arm/silabs_exx32/efm32pg1b/Kconfig.series]: Thank you. I think you're right. I've fixed this.

**Review by fkokosinski — APPROVED** [review]

**Review by DRuffer-tmo — APPROVED** [review]

**Comment by wsipak** [comment]
ping @anangl 
Could you please review this as the ADC maintainer? Thank you.

**Review by tgorochowik — APPROVED** [review]

**Pull request merged**